src/EStimLibrary/Core/ResourceManager.cs

```diff
@@ -8,13 +8,15 @@ public class ResourceManager<ResourceType>
 
     public readonly int MaxNumResources;
 
-    public int NumTotalResources => this.IdPool.NumUsedIds;
+    public int NumTotalResources => this.Resources.Count;
 
+    // Constructor
```

Comment:
Make this a docstring comment, i.e., ///, etc
Also, include docstring comments for all classes, properties, fields, and methods